source/administration/upgrade.rst

```diff
@@ -97,13 +97,6 @@ Location of your local storage directory
 
      sudo find mattermost/ mattermost/client/ -mindepth 1 -maxdepth 1 \! \( -type d \( -path mattermost/client -o -path mattermost/client/plugins -o -path mattermost/config -o -path mattermost/logs -o -path mattermost/plugins -o -path mattermost/data \) -prune \) | sort | sudo xargs rm -r
     
-#. Rename the ``plugins`` directories so they do not interfere with the upgrade.

```

Comment:
The intent here is to relocate unmanaged plugins out of the way before reinstating them later on, e.g. see line 149:
```
#. Reinstate the ``plugins`` directories, then restart the mattermost service.
```

Unmanaged plugins are plugins are relatively rare: most plugins are prepackaged or managed in the filestore, with the `plugins` directory being a "working space" for both. However, it remains possible to manually put a plugin into the `plugins` directory and have the server consume /that/.

Perhaps this is something we might change for v6.0, and just deprecate support for unmanaged plugins altogether.

Reply:
If an admin leaves `plugins` intact, the plugins directory gets clobbered right?

Reply:
Ah, that's a good observation @ali-farooq0 -- in fact, I think to @gigawhitlocks's point, unmanaged plugins in this directory should remain untouched through the upgrade. So maybe we just remove the additional step at the end re: `Reinstate the plugins directories` and everything will be fine?

Reply:
Unmanaged plugins remain untouched and you don't need to move or reinstate them. That's what I observed in my testing/usage.

Reply:
The plugin directories are omitted in the `find` command and then `mv` is later done with no clobber so I was able to do an upgrade and just leave my unmanaged plugins in place.

Reply:
> deprecate support for unmanaged plugins altogether.

I hope this doesn't mean deprecate support for custom unpublished plugins :P

Reply:
> I hope this doesn't mean deprecate support for custom unpublished plugins :P

No, definitely not -- just the "installation" strategy for same. @gigawhitlocks, if you want to remove the other half of this instruction from the `Reinstate the ``plugins`` directories` line, I think we're a go.

Reply:
Excellent, done and done.